compiler/codegen/NodeEvaluation.cpp

```diff
@@ -295,7 +295,11 @@ OMR::CodeGenerator::whichChildToEvaluate(TR::Node * node)
          }
       }
 
-   node->setEvaluationPriority(nodePriority);
+   if (!node->getOpCode().isTreeTop())
```

Comment:
May I ask you to add a comment indicating why the evaluation priority need not be set for a tree-top node?  Perhaps stating that evaluation priority is only meaningful for nodes that yield values, and tree-top nodes do not yield values.